libbpf-rs/src/object.rs

```diff
@@ -329,7 +327,9 @@ impl Object {
 
             // Get the map fd
             let fd = unsafe { libbpf_sys::bpf_map__fd(next_ptr) };
-            if fd < 0 {
+            // Old kernels can have uninitialized internal maps:
+            // https://github.com/libbpf/libbpf/blob/5c31bcf220f66e70f39fd141f5b0c55c6ab65e8e/src/libbpf.c#L5009-L5022
+            if fd < 0 && !unsafe { libbpf_sys::bpf_map__is_internal(next_ptr) } {
                 return Err(Error::System(-fd));
```

Comment:
as @chenhengqi  mentioned, if object load succeeded, then map can't have fd < 0, except if that map wasn't created *due to supported reasons*. Right now the only case when map is not created is `.rodata.*` that's not used on old kernels that don't support such maps. But I'm planning on adding a feature that will allow to opt-out map loading by user. So we definitely shouldn't hardcode any assumption about is_internal_map, FEAT_GLOBAL_DATA, etc.

If map's fd is < 0, then we either should skip it or keep it with FD -1 (or whatever other means).